### PR TITLE
feat: transaction guard removal

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves transaction guard removal

## How this PR fixes it

The option to remove transaction guards via a removal modal has now been implemented into the settings.

## How to test it

Enable a transaction guard (DenyListGuard: `0x885dCFA4A720dBAEa1Dd3AB1bB7c31Ed5f04B1c6`) then remove it via the settings. The removal should succeed after execution and the guard no longer be active/visible in the settings.

## Analytics changes

Removal tracks a new "Remove transaction guard" event.

## Screenshots
image

image

remove guard

### CODE REVIEW

File: src/components/settings/TransactionGuards/RemoveGuard/steps/ReviewRemoveGuard.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import { useSafeSDK } from '@/hooks/coreSDK/safeCoreSDK'
+import { Errors, logError } from '@/services/exceptions'
+import { trackEvent, SETTINGS_EVENTS } from '@/services/analytics'
+
+export const ReviewRemoveGuard = ({ data, onSubmit }: { data: RemoveGuardData; onSubmit: (data: null) => void }) => {
+  const { safe } = useSafeInfo()
+  const sdk = useSafeSDK()
+
+  const [safeTx, safeTxError] = useAsync<SafeTransaction>(() => {
+    return sdk?.createDisableGuardTx(data.address).then((tx) => createTx(tx.data))
```
Comment (katspaugh) [inline src/components/settings/TransactionGuards/RemoveGuard/steps/ReviewRemoveGuard.tsx:20]: I would make an fn in txSender, like all the other settings functions.

File: src/components/transactions/TxDetails/TxData/DecodedData/ValueArray/index.tsx [hunk 2]
```diff
@@ -25,9 +25,8 @@ export const Value = ({ type, ...props }: ValueArrayProps): ReactElement => {
                 type,
                 ...props,
                 value: address,
-                key,
               }
-              return <Value {...newProps} />
+              return <Value key={key} {...newProps} />
```
Comment (iamacook) [inline src/components/transactions/TxDetails/TxData/DecodedData/ValueArray/index.tsx:29]: The linter was complaining.

**Review by katspaugh — APPROVED** [review]
🚀

**Review by germartinez — APPROVED** [review]

**Comment by francovenica** [comment]
LGTM 
I was able to enable/disable the guard just fine
The event to track it also works fine

**Pull request merged**